widgets/hwmontemp_linux.lua

```diff
@@ -0,0 +1,60 @@
+----------------------------------------------------------------
+--   Licensed under the GNU General Public License v2
+--   (C) 2019, Alexander Koch <lynix47@gmail.com>
+----------------------------------------------------------------
+
+-- environment
+local io = { popen = io.popen, open = io.open }
+local assert = assert
+local setmetatable = setmetatable
+
+-- sysfs prefix for hwmon devices
+local sys_hwmon = "/sys/class/hwmon/"
+-- cache table for hwmon device names
+local paths = {}
+
+-- transparently caching hwmon device name lookup
+function name_to_path(name)
+    if paths[name] then return paths[name] end
+
+    for sensor in io.popen("ls -1 " .. sys_hwmon):lines() do
+        local path = sys_hwmon .. sensor
+        local f = assert(io.open(path .. "/name", "r"))
+        local sname = f:read("*line")
+        f:close()
+        if sname == name then
```

Comment:
I'm so sorry that I missed this part (I guess the reason is I was staying up late studying for finals). Also as @mic92 said, there is [`helpers.pathtotable`](https://github.com/vicious-widgets/vicious/blob/6b4910e0d14f97edebd2ab43698e63b1152bfad9/helpers.lua#L96) that you can use instead of manually `ls`ing.

Reply:
```lua suggestion
        local sensors = helpers.pathtotable("/sys/class/hwmon")
        local i = 0
        while sensors["hwmon" .. i] do
            local s = "hwmon" .. i
            if sensors[s].name == name .. "\n" then
```
Doesn't seem much easier to me, but maybe I'm just not getting the intended way to use `helpers.pathtotable` ;)